### PR TITLE
[consensus] Add an integration test for testing when start up requires a state sync

### PR DESCRIPTION
**Pull request by ankushagarwal**
## Motivation

Add integration (smoke) test for testing when start up requires a state sync

At some point there was a problem when component startup order was such that startup synchronization procedure was broken in real environment, while all unit tests were passing(because those test don't use same startup sequence as real node).

More details in https://github.com/libra/libra/issues/766

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

This is a test. Ran `cargo test test_startup_sync_state`

Closes https://github.com/libra/libra/issues/766

### CODE REVIEW

**Comment by ankushagarwal** [comment]
cc @andll @dmitri-perelman

File: testsuite/tests/libratest/smoke_test.rs [hunk 1]
```diff
@@ -322,6 +323,70 @@ fn test_basic_restartability() {
...
+    assert_eq!(
+        Decimal::from_f64(90.0),
+        Decimal::from_str(&client_proxy.get_balance(&["b", "0"]).unwrap()).ok()
+    );
+    assert_eq!(
+        Decimal::from_f64(10.0),
+        Decimal::from_str(&client_proxy.get_balance(&["b", "1"]).unwrap()).ok()
+    );
+    let peer_to_stop = 0;
+    // TODO: Can this the client lose its connection and be unable to recover
```
Comment (andll) [inline testsuite/tests/libratest/smoke_test.rs]: I think it should be fine

**Comment by zekun000** [comment]
lgtm

**Comment by ankushagarwal** [comment]
> Also, can you add panic locally in chained_bft_smr.rs after if initial_data.need_sync() { and confirm that (a) It fails this smoke test (b) Does not fail other smoke tests

Verified that this only fails `test_startup_sync_state`

```
---- smoke_test::test_startup_sync_state stdout ----
Base directory containing logs and configs: Temporary(TempPath("/var/folders/pl/fwpsbmh93mj444lfn6xpyf800000gn/T/3e88a3f7fc6901e60d10f66edd268218"))
waiting ......................transaction is stored!
waiting ......................transaction is stored!
thread 'smoke_test::test_startup_sync_state' panicked at 'called `Result::unwrap()` on an `Err` value: RpcFailure(RpcStatus { status: RpcStatusCode(4), details: Some("Deadline Exceeded") })', src/libcore/result.rs:1165:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
logs located at "/var/folders/pl/fwpsbmh93mj444lfn6xpyf800000gn/T/3e88a3f7fc6901e60d10f66edd268218"


failures:
    smoke_test::test_startup_sync_state

test result: FAILED. 10 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out
```

**Comment by andll** [comment]
Thanks!
One last small thing, can you add `[consensus]` in the beginning of first line of commit message?
https://github.com/libra/libra/blob/master/CONTRIBUTING.md#meaningful-commit-messages

**Comment by andll** [comment]
@bors-libra r+

Thanks @ankushagarwal . Great first diff :)

**Pull request closed without merging**